### PR TITLE
Use Nuget for VSSDK Packages and upgrade to stable package versions

### PR DESCRIPTION
**Pull request by cloudRoutine**
- updated VSSDK packages and build tools to stable version `15.0.26201`
- all VSSDK packages are retrieved from nuget, installing the VSSDK should no longer be necessary for the repo
- updated Microsoft.CodeAnalysis packages to stable version '2.0.0'

I started out trying to resolve some package conflicts between `Microsoft.CodeAnalysis` dependencies so I could get to work on building some custom workspaces just for testing, but once I got started I figured I might as well do the whole thing.

### CODE REVIEW

**Comment by majocha** [comment]
Build debug fails... Hope it's not too good to be true :(

**Comment by cloudRoutine** [comment]
it's possible the config is slightly off and I was inadvertently relying on some packages that aren't getting tracked or restored properly

i'll investigate later today

**Comment by majocha** [comment]
I did `git clean -fdx` first.
Perl installed for quite a while. But I tried once again and same.

File: packages.config [hunk 1]
```diff
@@ -27,13 +28,13 @@
   <package id="BenchmarkDotNet" version="0.9.8"/>
   <package id="BenchmarkDotNet.Diagnostics.Windows" version="0.9.8"/>
   <package id="VisualCppTools" version="14.0.24519-Pre"/>
-  <package id="Newtonsoft.Json" version="6.0.8"/>
+  <package id="Newtonsoft.Json" version="8.0.1"/>
```
Comment (majocha) [inline packages.config]: Why not 10.0.1?

File: vsintegration/packages.config [hunk 2]
```diff
@@ -1,29 +1,70 @@
...
+  <package id="Microsoft.VisualStudio.CoreUtility" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Shell.Design" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Shell.Framework" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Shell.15.0" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Settings.15.0" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Utilities" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Language.StandardClassification" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Language.Intellisense" version="15.0.26201" targetFramework="net46" />
+  <package id="Microsoft.VSSDK.BuildTools" version="15.0.26201" />
+15.0.26201-alpha
```
Comment (saul) [inline vsintegration/packages.config]: This looks out of place?

**Comment by saul** [comment]
If this is true, can we remove the "VS SDK" requirements from the readme as well in this PR? 

Thanks :)

**Comment by majocha** [comment]
This should work in principle. `extern alias` is a great idea.
With this we can finally use options handling code from Roslyn and have a nicely integrated XAML based options page.

File: packages.config (hunk 1 above)
Comment (cloudRoutine) [inline packages.config]: i was consistently getting warning messages during the build that this specific version couldn't be found

**Comment by cloudRoutine** [comment]
I deleted my packages and the proto compiler and builds stopped working for me too

**Comment by majocha** [comment]
@cloudRoutine just getting Roslyn and VS SDK up to the current version would be immediately useful.

**Comment by cloudRoutine** [comment]
@majocha looks like i just forgot to update the proto compiler fsproj, give it another shot?

File: vsintegration/packages.config (hunk 2 above)
Comment (cloudRoutine) [inline vsintegration/packages.config]: whoops, good catch

**Comment by majocha** [comment]
Still something wrong. I know it can be done!

**Comment by cloudRoutine** [comment]
@majocha can you put the errors/output log in a gist?

**Comment by majocha** [comment]
[https://gist.github.com/majocha/7e565df2d0d84a24d5606a0165a521ff](https://gist.github.com/majocha/7e565df2d0d84a24d5606a0165a521ff)

**Comment by cloudRoutine** [comment]
@majocha give it another spin?

**Comment by cloudRoutine** [comment]
@KevinRansom I've taken care of everything I wanted to now. The UnitTests lib has the references I need to start working on a custom workspace without the conflicts I ran into the first time I tried to work on it.

@smoothdeveloper could you try out this PR? I'm curious if it'll help with the issues you seem to keep having with building VFT.

**Comment by majocha** [comment]
Yes!

And now it also builds with less warnings than master.
Awesome :)

**Comment by majocha** [comment]
And now it's easy to implement:

**Comment by vasily-kirichenko** [comment]
@majocha wow! :)

**Comment by majocha** [comment]
@vasily-kirichenko not wired up yet... Just a test if it shows up correctly. A lot of mucking with localized string resources to do it properly. But we can copy paste XAML from Roslyn, change it a bit and load it. @cloudRoutine great job 👍

**Comment by KevinRansom** [comment]
@cloudRoutine 
Thanks for taking care of this

Kevin

**Pull request merged**